### ISSUE TITLE
Improve extented ctags

### ISSUE THREAD

**Issue by liuchengxu** [liuchengxu/vista.vim#38]
follow-up problems of #35 :

- [ ] how to take care of the custom ctags cmd.
- [ ] transform the short kind to the long, e.g., the long kind missing in hasktags.
- [ ] how to handle multiple filetypes share one type, e.g., cpp and cuda.
- [ ] the new ctags options may be unneccessary for the finder functionality, which can be replaced with the previous simpler one.

**Comment by leoatchina** [comment]
how to set `kinds` for ctags?
for example, when i use tagbar and use `tstags` for  typescript,  the following config works good with `Univeral ctags`
```
                if executable('tstags')
                    let g:tagbar_type_typescript = {
                        \ 'ctagstype': 'typescript',
                        \ 'ctagsbin':  'tstags',
                        \ 'ctagsargs': '-f-',
                        \ 'sort' : 0,
                        \ 'kinds': [
                            \ 'e:enums:0:1',
                            \ 'f:function:0:1',
                            \ 't:typealias:0:1',
                            \ 'M:Module:0:1',
                            \ 'I:import:0:1',
                            \ 'i:interface:0:1',
                            \ 'C:class:0:1',
                            \ 'm:method:0:1',
                            \ 'p:property:0:1',
                            \ 'v:variable:0:1',
                            \ 'c:const:0:1',
                        \ ]
                        \ }
                endif
```
when use vista.vim, i can config like this
```
            if executable('tstags')
                let g:vista_ctags_cmd = {
                    \ 'typescript': 'tstags -f-',  # -f- is important 
                    \ }
            endif
```
is there any way to config `kinds`  like tagbar

**Comment by liuchengxu** [comment]
It's blocked by #30. vista.vim does not manipulate kinds at present.

**Comment by liuchengxu** [comment]
@leoatchina I currently don't have a plan to manipulate kinds after mostly finishing #30 as I am not aware of its advantage. If you really want this, you can raise another issue, maybe others are interested  as well.
